synapse/rest/client/devices.py

```diff
@@ -80,35 +83,37 @@ def __init__(self, hs: "HomeServer"):
         self.device_handler = hs.get_device_handler()
         self.auth_handler = hs.get_auth_handler()
 
+    class PostBody(RequestBodyModel):
+        auth: Optional[AuthenticationData]
+        devices: List[StrictStr]
+
     @interactive_auth_handler
     async def on_POST(self, request: SynapseRequest) -> Tuple[int, JsonDict]:
         requester = await self.auth.get_user_by_req(request)
 
         try:
-            body = parse_json_object_from_request(request)
+            body = parse_and_validate_json_object_from_request(request, self.PostBody)
         except errors.SynapseError as e:
             if e.errcode == errors.Codes.NOT_JSON:
-                # DELETE
+                # TODO: Can/should we remove this fallback now?
```

Comment:
wonder if the haproxy logs include the count of bytes in the request body. If so, might be able to see if there are still any clients relying on this ...

Reply:
The [httplog format](https://www.haproxy.com/documentation/hapee/1-8r1/onepage/#8.2.3) (which we use) doesn't include this. But there is a [field we could opt into](https://www.haproxy.com/documentation/hapee/1-8r1/onepage/#8.2.4):

> `  |   | %U   | bytes_uploaded       (from client to server)  | numeric     |`

I would guess this includes the HTTP request line and headers: would have  to check.